dlib/cuda/cuda_dlib.h

```diff
@@ -423,7 +423,6 @@ namespace dlib
             compute_loss_multiclass_log_per_pixel(
             )
             {
-                work = device_global_buffer();
```

Comment:
Oh yeah, good catch, this is definitely the issue as this pins the buffer to a specific device.